### ISSUE TITLE
[Vanilla, NPC, Spell] Deadwood Shaman instantly casts Chain Lightning/Curse of the Deadwood instead of on a Cast Time; casts them during other ongoing Spells as well; don't register on the Combat Log; missing Spell School lockout to them from Interrupts

### ISSUE THREAD

**Issue by Aridam** [SunwellTracker/issues#2434]
Decription:
Deadwood Shamen cast their Chain Lightnings instantly rather than on a Cast Time, while casting something else, and they don't register as a cast.

How it works:
Deadwood Shamen in northern Felwood instantly cast Chain Lightnings.
Furthermore, their CLs happen while they're casting something else, like Lightning Bolt/Curse of the Deadwood.
Lastly, their casts of CL don't register in the Combat Log -- only their Damage.

18:15:01> Deadwood Shaman begins casting [Lightning Bolt]. 
18:15:01> Deadwood Shaman's [Chain Lightning] hits Shroudsoul for 344 Nature. 

There is no "Deadwood Shaman begins casting [Chain Lightning]" -- just jumps straight to the Damage.
Also as you can see, they begin casting LB and immediately with it, just randomly zap you with an instant CL.

How it should work:
Their Chain Lightnings should be on a 3s Cast Time.
They should also be unable to use them while casting something else.
Their casts of CL should register in the Combat Log, rather than just its DMG.

Steps to reproduce:
1. Go to Felpaw Village in northern Felwood;
2. Fight a Deadwood Shaman;
3. Observe them using CL instantly, even while casting something else, whilst the Combat Log doesn't register the cast attempts but rather the DMG it does.

NPC ID: https://db.darkwizard.pl/?npc=7158#?ref=sunwell.pl
NPC's Chain Lightning ID: https://db.darkwizard.pl/?spell=12058

**Comment by Aridam** [comment]
Apparently their Curse of the Deadwood is the same -- instant cast, done during other casts, attempts don't register in the Combat Log other than being afflicted by it.

18:36:01> Deadwood Shaman begins casting [Lightning Bolt]. 
18:36:02> Shroudsoul is afflicted by Deadwood Shaman's [Curse of the Deadwood]. 
Instantly-cast and unregistered as attempted during the LB cast.
Supposed to be on a 2s Cast Time and register as an extra line saying "begins casting [Curse of the Deadwood]"

Steps to reproduce are the same as with CL above.

NPC's CotD ID: https://db.darkwizard.pl/?spell=13583

**Comment by Aridam** [comment]
Another update: I just got hit by both at the same time _while Interrupted_ from my Pummel within like a second of that happening. CL shouldn't have been possible because I used Pummel on LB -- both are Nature-type Spells, so if anything only CotD should've hit as it's Shadow-type.

Steps to reproduce missing Spell School lockout:
1. Interrupt Lightning Bolt;
2. Watch for possible Chain Lightning casts done within the Interrupt's lockout period.
